### ISSUE TITLE
Any workaround on deploy these models using TensorRT?

### ISSUE THREAD

**Issue by jinfagang** [facebookresearch/detectron2#192]
Since the model is really nice, it would be very good to add a feature deploy on TensorRT, does there anyone in community has some walk around on this?

**Comment by jinyeom** [comment]
I think the best way to deploy a model at the moment is to take the backbone and heads out of it and make a separate `nn.Module` with them. The built-in backbones don't seem to convert to ONNX nicely out of the box though -- I'd recommend using `torchvision` backbones instead.

**Comment by jinfagang** [comment]
@jinyeom  You mean using TensorRT on backbone only? That's actually doesn't accelerate much. and even wastes lots of time copying data.

**Comment by jinyeom** [comment]
Sorry, I meant you should get the backbone+heads out of the original model and build a separate module. All the other stuff like NMS and anchor generation don't convert.
Check this repo for reference: https://github.com/NVIDIA/retinanet-examples

**Comment by jinfagang** [comment]
@jinyeom  Good advise. Have u tried this way?

**Comment by jinfagang** [comment]
@jinyeom  I have digged a while, it hard to do since all layer input inside detectron2 using ImageList object not barely python list or tuple which is massive workload to change them.

The roi_heads using output of ProposalGenerator, while proposal using ImageList as input, so it's impossible trace only backbone and roi_heads since you gonna need ProposalGenerator output. But that part hugely relies on ImageList provides every image height and width information.

**Comment by jinyeom** [comment]
I'm still exploring this method at the moment. Yeah, you're probably going to have to reimplement that preprocessing part manually.

**Comment by jinfagang** [comment]
@jinyeom  I have found it possible with ImageList when trace. previous maskrcnn-benchmark had people exported onnx by doing this.

However, I got some constant error for now.. needs fix.

**Comment by ppwwyyxx** [comment]
We will not have GPU inference pipeline since it has low value to facebook. Closing but welcome to comment about any working community efforts .

**Comment by Sanster** [comment]
Just share some test result. I converted backbone to TensorRT engine with [torch2trt](https://github.com/NVIDIA-AI-IOT/torch2trt). If only measure backbone forward time, TensorRT engine is 30% faster, but the while `GeneralizedRCNN` forward only 5%~10% faster.

**Comment by automata** [comment]
@Sanster that's great. Are you willing to share the steps to convert backbone to TensorRT? Are you planning to convert whole Mask RCNN to TRT? Thanks!

**Comment by adizhol** [comment]
Has anyone tried to convert the Deformable Convoultion layer(s) to TenosrRT or ONNX?

**Comment by mpjlu** [comment]
I have tried to use TensorRT to boost the inference performance of Cascade R-CNN with Caffe2/tensorrt. Caffe2/tensorrt is like TF-TRT, but Caffe2/tensorrt is not mature enough，after fixing some bugs, I got it run. https://zhuanlan.zhihu.com/p/122399743

**Comment by jinfagang** [comment]
@mpjlu  Can u share some code to reproduce runing Cascaed RCNN on trt?

**Comment by ysyyork** [comment]
> Just share some test result. I converted backbone to TensorRT engine with [torch2trt](https://github.com/NVIDIA-AI-IOT/torch2trt). If only measure backbone forward time, TensorRT engine is 30% faster, but the while `GeneralizedRCNN` forward only 15% faster.
> 
> * config-file: faster_rcnn_R_101_FPN_3x.yaml
> * GPU: 1080
> * TensorRT: 6.0.1.5
> * pytorch: 1.3.0

can you sure some code snippet? i can't make your example work with torch2trt
